src/Pim/Component/Catalog/Model/AbstractProduct.php

```diff
@@ -667,4 +671,24 @@ public function getReference()
     {
         return $this->getIdentifier();
     }
+
+    /**
+     * @return ArrayCollection
+     */
+    public function getUniqueData()
```

Comment:
imho, it's not confusing
data is always plural
the problem is just that we are really good in English

but I understand your point
and I renamed all the usages inside methods to make it more clear